docs/reference/style-guide.rst

```diff
@@ -0,0 +1,392 @@
+Ubuntu Package Guide -- Style Guide
+===================================
+
+.. note::
+
+    This Style Guide is based on `Canonical' reStructuredText style guide`_.
+
+All documentation files should use `reStructuredText`_ (reST) syntax. Community
+contributions in `markdown`_ are welcome, but they should be reformated to reST, before
+they are merged into the main branch 
+
+For general style conventions, see the `Canonical Documentation Style Guide`_.
+
+Headings
+--------
+
+.. grid:: 2
+    
+    .. grid-item::
+
+        .. code:: reStructuredText
+
+            Title
+            =====
+
+    .. grid-item::
+        :child-align: center
+
+        Page title and ``<h1>`` heading
+
+    .. grid-item::
+
+        .. code:: reStructuredText
+
+            Heading
+            -------   
+            
+    .. grid-item::
+        :child-align: center
+
+        ``<h2>`` heading
+    
+    .. grid-item::
+
+        .. code:: reStructuredText
+
+            Heading
+            ~~~~~~~   
+            
+    .. grid-item::
+        :child-align: center
+
+        ``<h3>`` heading
+
+    .. grid-item::
+
+        .. code:: reStructuredText
+
+            Heading
+            ^^^^^^^
+            
+    .. grid-item::
+        :child-align: center
+
+        ``<h4>`` heading        
+
+    .. grid-item::
+
+        .. code:: reStructuredText
+
+            Heading
+            .......   
+            
+
+    .. grid-item::
+        :child-align: center
+
+        ``<h5>`` heading
+
+Underlines must be the same length as the title or heading.
+
+Adhere to the following conventions:
+
+- Do not use consecutive headings without intervening text.
+- Be consistent with the characters you use for each level. Use the ones specified above.
+- Do not skip levels (for example, do not follow an H2 heading with an H4 heading).
+- Use sentence style for headings (capitalise only the first word).
```

Comment:
I'm not a fan of the "negative" phrasing employed in the original, and would be happy if we avoided it here. For example, instead of saying "Do not use consecutive headings without intervening text" we could use "Make sure that you include text between consecutive headings" and so on. It sounds less stark and a bit more welcoming to phrase it in a more positive way of "you should do this" vs. "don't do that".